src/types/UiaTextRangeBase.cpp

```diff
@@ -425,23 +383,65 @@ IFACEMETHODIMP UiaTextRangeBase::GetBoundingRectangles(_Outptr_result_maybenull_
         // order: left, top, width, height. each line will have its own
         // set of coords.
         std::vector<double> coords;
-        const TextBufferRow startRow = _endpointToTextBufferRow(_pData, _start);
 
-        if (_degenerate && _isScreenInfoRowInViewport(_pData, startRow))
+        const auto bufferSize = _pData->GetTextBuffer().GetSize();
+
+        // these viewport vars are converted to the buffer coordinate space
+        const auto viewport = bufferSize.ConvertToOrigin(_pData->GetViewport());
+        const auto viewportOrigin = viewport.Origin();
+        const auto viewportEnd = viewport.EndInclusive();
+
+        // startAnchor: the earliest COORD we will get a bounding rect for
+        auto startAnchor = GetEndpoint(TextPatternRangeEndpoint_Start);
+        if (bufferSize.CompareInBounds(startAnchor, viewportOrigin, true) < 0)
+        {
+            // earliest we can be is the origin
+            startAnchor = viewportOrigin;
+        }
+
+        // endAnchor: the latest COORD we will get a bounding rect for
+        auto endAnchor = GetEndpoint(TextPatternRangeEndpoint_End);
+        if (bufferSize.CompareInBounds(endAnchor, viewportEnd, true) > 0)
+        {
+            // latest we can be is the viewport end
+            endAnchor = viewportEnd;
+        }

```

Comment:
This doesn't inhibit someone from trying to move to a position that is "scrolled up" or "scrolled down" from the current viewport? Is this method only for the graphical presentation of the overlay rectangle that highlights the range?

Reply:
Correct. This is what draws these yellow boxes:
![image](https://user-images.githubusercontent.com/11050425/73424499-d9f5de80-42e3-11ea-84e2-2420babe6995.png)